discovery-provider/src/tasks/users.py

```diff
@@ -523,19 +523,29 @@ def update_user_events(
             # There is something wrong with events, don't process it
             return
 
-        # Mark existing UserEvents entries as not current
-        session.query(UserEvents).filter_by(
-            user_id=user_record.user_id, is_current=True
-        ).update({"is_current": False})
+        # Get existing UserEvents entry
+        existing_user_events = (
+            session.query(UserEvents)
+            .filter_by(user_id=user_record.user_id, is_current=True)
+            .one_or_none()
+        )
 
         user_events = UserEvents(
             user_id=user_record.user_id,
             is_current=True,
             blocknumber=user_record.blocknumber,
             blockhash=user_record.blockhash,
+            referrer=existing_user_events.referrer if existing_user_events else None,
```

Comment:
super optional nit, but the ternary over multiple lines is a little confusing to read as a kwarg, maybe 

existing_referrer = existing_user_events.referrer if existing_user_events else None
existing_is_mobile_user = existing_user_events.is_mobile_user if existing_user_events else False

user_events = UserEvents(
  referrer=existing_referrer,
  is_mobile_user=existing_is_mobile_user
)